sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -420,6 +420,18 @@ private[spark] object SQLConf {
   val USE_SQL_AGGREGATE2 = booleanConf("spark.sql.useAggregate2",
     defaultValue = Some(true), doc = "<TODO>")
 
+  val COMBINE_SMALL_FILE = booleanConf(
+    "spark.sql.small.file.combine",
+    defaultValue = Some(true),
+    isPublic = false
+  )
```

Comment:
This option is made private and default to true, which assumes that reading small files is the common case, which is neither true nor recommended.